src/nirfsg/system_tests/test_system_nirfsg.py

```diff
@@ -0,0 +1,559 @@
+        test_rfsg_device_session.write_arb_waveform('mywaveform1', waveform_data, False)
+        markerlocations = [1, 100, 200]
+        test_rfsg_device_session.waveform['mywaveform1'].markers[0].set_waveform_marker_event_locations(markerlocations)
+        markerlocations_out = test_rfsg_device_session.waveform['mywaveform1'].markers[0].get_waveform_marker_event_locations()
+        assert markerlocations_out == markerlocations
+
+    def test_write_script(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_generation_mode(nirfsg.GenerationMode.SCRIPT)
+        waveform_data = np.full(1000, 0.707 + 0.707j, dtype=np.complex64)
+        test_rfsg_device_session.write_arb_waveform('mywaveform1', waveform_data, False)
+        script = '''script myScript1
+        repeat forever
+        generate mywaveform1
+        end repeat
+        end script'''
+        test_rfsg_device_session.write_script(script)
+
+    @pytest.mark.skipif(use_simulated_session is True, reason="Scripts not compiled on simulated device")
+    def test_check_if_script_exists(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_generation_mode(nirfsg.GenerationMode.SCRIPT)
+        waveform_data = np.full(1000, 0.707 + 0.707j, dtype=np.complex64)
+        test_rfsg_device_session.write_arb_waveform('mywaveform1', waveform_data, False)
+        script = '''script myScript1
+        repeat forever
+        generate mywaveform1
+        end repeat
+        end script'''
+        test_rfsg_device_session.write_script(script)
+        script_exists = test_rfsg_device_session.check_if_script_exists('myScript1')
+        assert script_exists is True
+        script_exists = test_rfsg_device_session.check_if_script_exists('myScript2')
+        assert script_exists is False
+
+    @pytest.mark.skipif(use_simulated_session is True, reason="Scripts not compiled on simulated device")
+    def test_write_script_with_bad_script(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_generation_mode(nirfsg.GenerationMode.SCRIPT)
+        script = '''script myScript1
+        repeat forever
+        generate mywaveform1
+        end repeat
+        end script'''
+        try:
+            test_rfsg_device_session.write_script(script)
+            assert False
+        except nirfsg.Error as e:
+            assert e.code == -1074101603
+            assert "A waveform matching the provided name does not exist in memory" in e.description
+
+    def test_configure_software_trigger(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_software_start_trigger()
+        assert test_rfsg_device_session.start_trigger_type == nirfsg.StartTrigType.SOFTWARE
+        test_rfsg_device_session.configure_software_script_trigger('scriptTrigger0')
+        assert test_rfsg_device_session.script_triggers[0].script_trigger_type == nirfsg.ScriptTrigType.SOFTWARE
```

Comment:
Given the existence of the `start_trigger_type` and `script_trigger_type`, `configure_software_start_trigger`, `configure_software_script_trigger`, `disable_start_trigger` and `disable_script_trigger` all seem to be of questionable value.

However, I don't feel as strongly about these as I do `configure_generation_mode` and `configure_output_enabled`, given that you may just want them to round out the API, due to the existence of `configure_digital_edge_start_trigger` and `configure_digital_edge_script_trigger`.

@marcoskirsch may feel more strongly about it than I do.